src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/resthandler/RestChangePolicyAction.kt

```diff
@@ -0,0 +1,244 @@
+import org.elasticsearch.action.admin.cluster.state.ClusterStateResponse
+import org.elasticsearch.action.bulk.BulkRequest
+import org.elasticsearch.action.bulk.BulkResponse
+import org.elasticsearch.action.search.SearchRequest
+import org.elasticsearch.action.search.SearchResponse
+import org.elasticsearch.action.support.IndicesOptions
+import org.elasticsearch.client.node.NodeClient
+import org.elasticsearch.cluster.service.ClusterService
+import org.elasticsearch.common.Strings
+import org.elasticsearch.common.bytes.BytesReference
+import org.elasticsearch.common.settings.Settings
+import org.elasticsearch.common.xcontent.LoggingDeprecationHandler
+import org.elasticsearch.common.xcontent.NamedXContentRegistry
+import org.elasticsearch.common.xcontent.XContentHelper
+import org.elasticsearch.common.xcontent.XContentParser
+import org.elasticsearch.common.xcontent.XContentParser.Token
+import org.elasticsearch.common.xcontent.XContentParserUtils.ensureExpectedToken
+import org.elasticsearch.common.xcontent.XContentType
+import org.elasticsearch.index.query.IdsQueryBuilder
+import org.elasticsearch.rest.BaseRestHandler
+import org.elasticsearch.rest.BytesRestResponse
+import org.elasticsearch.rest.RestChannel
+import org.elasticsearch.rest.RestController
+import org.elasticsearch.rest.RestRequest
+import org.elasticsearch.rest.RestRequest.Method.POST
+import org.elasticsearch.rest.RestResponse
+import org.elasticsearch.rest.RestStatus
+import org.elasticsearch.rest.action.RestResponseListener
+import org.elasticsearch.search.builder.SearchSourceBuilder
+import java.io.IOException
+
+class RestChangePolicyAction(
+        settings: Settings,
+        controller: RestController,
+        val clusterService: ClusterService
+) : BaseRestHandler(settings) {
+
+    private val log = LogManager.getLogger(javaClass)
+
+    companion object {
+        const val CHANGE_POLICY_BASE_URI = "$ISM_BASE_URI/change_policy"
+        const val INDEX_NOT_MANAGED = "This index is not being managed"
+        const val INDEX_IN_TRANSITION = "This index is already transitioning"
+        const val INDEX_NOT_INITIALIZED = "This managed index has not been initialized yet"
+    }
+
+    init {
+        controller.registerHandler(POST, "$ISM_BASE_URI/change_policy", this)
+        controller.registerHandler(POST, "$ISM_BASE_URI/change_policy/{index}", this)
+    }
+
+    override fun getName(): String = "change_policy_action"
+
+    @Throws(IOException::class)
+    override fun prepareRequest(request: RestRequest, client: NodeClient): RestChannelConsumer {
+        val indices: Array<String>? = Strings.splitStringByCommaToArray(request.param("index"))
+        if (indices == null || indices.isEmpty()) {
+            throw IllegalArgumentException("Missing index")
+        }
+
+        val xcp = request.contentParser()
+        ensureExpectedToken(Token.START_OBJECT, xcp.nextToken(), xcp::getTokenLocation)
+        val changePolicy = ChangePolicy.parse(xcp)
```

Comment:
What kind of error does this `parse` throw if user input is incorrect. Does it give similar output as other ES requests?

Reply:
Should throw whatever error is thrown in parse.

```
 curl -X POST localhost:9200/_opendistro/_ism/change_policy/some_index -H 'Content-Type: application/json' -d '{ "invalid": "invalid" }'

{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"Invalid field: [invalid] found in ChangePolicy."}],"type":"illegal_argument_exception","reason":"Invalid field: [invalid] found in ChangePolicy."},"status":400}


curl -X POST localhost:9200/_opendistro/_ism/change_policy/some_index -H 'Content-Type: application/json' -d '{}'

{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"ChangePolicy policy id is null"}],"type":"illegal_argument_exception","reason":"ChangePolicy policy id is null"},"status":400}
```